domains/primitives/messenger-host-functions/src/host_functions.rs

```diff
@@ -70,14 +70,32 @@ where
             .header(consensus_block_hash)
             .ok()
             .flatten()?;
+
+        // if we are snap syncing, it is possible that block's parent may not exist.
+        // in such cases, we would rather just pick the best hash to fetch the domain runtime
+        // This is fine to fetch the domain code since we only use it to derive the storage key.
+        // This will not be an issue unless the Storage type was changed for Outbox and Inbox Response.
+        // which might not happen but good to be aware if such situations arise.
```

Comment:
Do we need a test to make sure the storage type hasn't changed?

Reply:
not required. If the storage type changes so does the api call. We would then need to have api version check and derive key accordingly

Reply:
BTW The host function is costly and may be unnecessary, see https://github.com/autonomys/subspace/issues/3486